cvat/apps/engine/task.py

```diff
@@ -319,68 +321,101 @@ def update_progress(progress):
     video_path = ""
     video_size = (0, 0)
 
+    def _update_status(msg):
+        job.meta['status'] = msg
+        job.save_meta()
+
     if settings.USE_CACHE and db_data.storage_method == StorageMethodChoice.CACHE:
        for media_type, media_files in media.items():
 
             if not media_files:
                 continue
 
+            # replace manifest file (e.g was uploaded 'subdir/manifest.jsonl')
+            if manifest_file and not os.path.exists(db_data.get_manifest_path()):
+                shutil.copyfile(os.path.join(upload_dir, manifest_file[0]),
+                    db_data.get_manifest_path())
+                if upload_dir != settings.SHARE_ROOT:
+                    os.remove(os.path.join(upload_dir, manifest_file[0]))
+
             if task_mode == MEDIA_TYPES['video']['mode']:
                 try:
-                    if meta_info_file:
+                    if manifest_file:
                         try:
-                            from cvat.apps.engine.prepare import UploadedMeta
-                            meta_info = UploadedMeta(source_path=os.path.join(upload_dir, media_files[0]),
-                                                     meta_path=db_data.get_meta_path(),
-                                                     uploaded_meta=os.path.join(upload_dir, meta_info_file[0]))
-                            meta_info.check_seek_key_frames()
-                            meta_info.check_frames_numbers()
-                            meta_info.save_meta_info()
-                            assert len(meta_info.key_frames) > 0, 'No key frames.'
+                            manifest = VideoManifestValidator(source_path=os.path.join(upload_dir, media_files[0]),
+                                                              manifest_path=db_data.get_manifest_path())
+                            manifest.init_index()
+                            manifest.validate_seek_key_frames()
+                            manifest.validate_frame_numbers()
+                            assert len(manifest) > 0, 'No key frames.'
+
+                            all_frames = manifest['properties']['length']
+                            video_size = manifest['properties']['resolution']
                         except Exception as ex:
```

Comment:
I think catching `Exception` for such a broad code fragment can hide some implementation errors (according to the following check in the line 355). Maybe it's worth logging, if the error is not an `AssertionError`?